Resources/Storage/GFAL2_SRM2Storage.py

```diff
@@ -97,77 +96,121 @@ def getTransportURL( self, path, protocols = False ):
-      listProtocols.insert( listProtocols.index( 'root' ), 'xroot' )
+      listProtocols.insert(listProtocols.index('root'), 'xroot')
     elif 'xroot' in listProtocols and 'root' not in listProtocols:
-      listProtocols.insert( listProtocols.index( 'xroot' ) + 1, 'root' )
-
+      listProtocols.insert(listProtocols.index('xroot') + 1, 'root')
 
     if self.protocolParameters['Protocol'] in listProtocols:
       successful = {}
       failed = {}
       for url in urls:
-        if self.isURL( url )['Value']:
+        if self.isURL(url)['Value']:
           successful[url] = url
         else:
           failed[url] = 'getTransportURL: Failed to obtain turls.'
 
-      return S_OK( {'Successful' : successful, 'Failed' : failed} )
+      return S_OK({'Successful': successful, 'Failed': failed})
 
     for url in urls:
-      res = self.__getSingleTransportURL( url, listProtocols )
-      self.log.debug( 'res = %s' % res )
+      res = self.__getSingleTransportURL(url, listProtocols)
+      self.log.debug('res = %s' % res)
 
       if not res['OK']:
         failed[url] = res['Message']
       else:
         successful[url] = res['Value']
 
-    return S_OK( { 'Failed' : failed, 'Successful' : successful } )
-
-
+    return S_OK({'Failed': failed, 'Successful': successful})
 
-
-  def __getSingleTransportURL( self, path, protocols = False ):
+  def __getSingleTransportURL(self, path, protocols=False):
     """ Get the tURL from path with getxattr from gfal2
 
     :param self: self reference
     :param str path: path on the storage
     :returns: S_OK( Transport_URL ) in case of success
               S_ERROR( errStr ) in case of a failure
     """
-    self.log.debug( 'GFAL2_SRM2Storage.__getSingleTransportURL: trying to retrieve tURL for %s' % path )
+    self.log.debug(
+        'GFAL2_SRM2Storage.__getSingleTransportURL: trying to retrieve tURL for %s' %
+        path)
     if protocols:
-      self.ctx.set_opt_string_list( "SRM PLUGIN", "TURL_PROTOCOLS", protocols )
+      self.ctx.set_opt_string_list("SRM PLUGIN", "TURL_PROTOCOLS", protocols)
 
-    res = self._getExtendedAttributes( path, attributes = ['user.replicas'] )
+    res = self._getExtendedAttributes(path, attributes=['user.replicas'])
     self.__setSRMOptionsToDefault()
 
     if res['OK']:
-      return S_OK( res['Value']['user.replicas'] )
-    else:
-      errStr = 'GFAL2_SRM2Storage.__getSingleTransportURL: Extended attribute tURL is not set.'
-      self.log.debug( errStr, res['Message'] )
-      return res
+      return S_OK(res['Value']['user.replicas'])
+
+    errStr = 'GFAL2_SRM2Storage.__getSingleTransportURL: Extended attribute tURL is not set.'
+    self.log.debug(errStr, res['Message'])
+    return res
+
+  def getOccupancy(self, *parms, **kws):
+    """ Gets the GFAL2_SRM2Storage occupancy info.
+
+      TODO: needs gfal2.15 because of bugs:
+      https://its.cern.ch/jira/browse/DMC-979
+      https://its.cern.ch/jira/browse/DMC-977
```

Comment:
Aren't you already "going around" these 2 bugs in the lines below?

Reply:
Yes, why ?

Reply:
OK, just to confirm.